genesyscloud/idp_adfs/resource_genesyscloud_idp_adfs.go

```diff
@@ -0,0 +1,145 @@
+package idp_adfs
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v129/platformclientv2"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	"terraform-provider-genesyscloud/genesyscloud/util/constants"
+	"terraform-provider-genesyscloud/genesyscloud/util/lists"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+/*
+The resource_genesyscloud_idp_adfs.go contains all of the methods that perform the core logic for a resource.
+*/
+
+// getAllAuthIdpAdfs retrieves all of the idp adfs via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthIdpAdfss(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := newIdpAdfsProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	_, resp, err := proxy.getIdpAdfs(ctx)
+	if err != nil {
+		if util.IsStatus404(resp) {
+			// Don't export if config doesn't exist
+			return resources, nil
+		}
+		return nil, util.BuildAPIDiagnosticError("genesyscloud_idp_adfs", fmt.Sprintf("Failed to get IDP ADFS error: %s", err), resp)
```

Comment:
You can use `resourceName` here instead of the hard-coded string "genesyscloud_idp_adfs". Same with the other instances below